src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -166,6 +167,38 @@ def validate_interval(self, interval):
             raise serializers.ValidationError("Invalid interval")
         return interval
 
+    def validate_layout(self, layout):
+        if layout is None:
+            return None
+
+        STORE_KEYS = {
+            "x",
+            "y",
+            "w",
+            "h",
+            "min_w",
+            "max_w",
+            "min_h",
+            "max_h",
+        }
+        IGNORE_KEYS = {
+            "i",
+            "static",
+            "moved",
+        }
+        invalid_keys = set(layout.keys()) - (STORE_KEYS.union(IGNORE_KEYS))
+        if invalid_keys:
+            invalid_keys_str = ", ".join(invalid_keys)
+            raise serializers.ValidationError(f"Contains invalid keys: {invalid_keys_str}")
+
+        layout_to_store = {}
+        for key in layout.keys():
+            if key in STORE_KEYS:
+                if not isinstance(layout[key], int):
+                    raise serializers.ValidationError(f"Expected number for: {key}")
+                layout_to_store[key] = layout[key]
```

Comment:
nit: can we loop through the STORE_KEYS instead? Esp if we don't end up doing validating the layout field JSON/Dict we may not know what/how many keys are coming through in this layout dict, so probably better to loop through known keys.

```suggestion
        for key in STORE_KEYS:
            if layout.get(key) is None:
                continue

            if not isinstance(layout[key], int):
                raise serializers.ValidationError(f"Expected number for: {key}")
            layout_to_store[key] = layout[key]
```

Reply:
https://github.com/getsentry/sentry/pull/31050/files#diff-016a07238f9714dd8918a63f4930f5dec5109e0a20b1807ad5ed98075666e6bcR69

Thanks for the suggestion!